packages/geo/src/lib/datasource/shared/datasources/wfs-datasource.ts

```diff
@@ -2,139 +2,97 @@ import olSourceVector from 'ol/source/Vector';
-      dynamicUrl: downloadBaseUrl
-    });
+    const queryStringValuesDownload = formatWFSQueryString(this.options);
+    const downloadBaseUrl = queryStringValuesDownload.find(f => f.name === 'getfeature').value;
+    this.options.download = Object.assign({}, this.options.download, { dynamicUrl: downloadBaseUrl });
 
     return new olSourceVector({
       format: this.getFormatFromOptions(),
       overlaps: false,
-      url: (extent, resolution, proj) => {
-        const srsname = this.options.paramsWFS.srsName
-          ? 'srsname=' + this.options.paramsWFS.srsName
-          : 'srsname=' + proj.getCode();
-
-        let filters;
-        if (
-          (this.options as OgcFilterableDataSourceOptions).ogcFilters &&
-          (this.options as OgcFilterableDataSourceOptions).ogcFilters.enabled
-        ) {
-          filters = (this.options as OgcFilterableDataSourceOptions).ogcFilters.filters;
-        }
-        this.options.paramsWFS.xmlFilter = this.ogcFilterWriter.buildFilter(
-          filters,
-          extent,
-          proj,
-          this.options.paramsWFS.fieldNameGeometry
-        );
-
-        let baseUrl = `${url}?${baseWfsQuery}&${wfsVersion}&${featureTypes}&`;
-        baseUrl += `${outputFormat}&${srsname}&${maxFeatures}`;
-
-        const patternFilter = /(filter|bbox)=.*/gi;
-        if (patternFilter.test(this.options.paramsWFS.xmlFilter)) {
-          baseUrl += `&${this.options.paramsWFS.xmlFilter}`;
-        }
-
-        this.options.download = Object.assign({}, this.options.download, {
-          dynamicUrl: baseUrl
-        });
-
-        return baseUrl;
+      url: (extent, resolution, proj) => { return this.buildUrl(extent, proj, ogcFilters);
       },
       strategy: OlLoadingStrategy.bbox
     });
   }
 
+  private buildUrl(extent, proj, ogcFilters): string {
+    const paramsWFS = this.options.paramsWFS;
+    const queryStringValues = formatWFSQueryString(this.options, undefined, proj.getCode());
+    let igoFilters;
+    if (ogcFilters && ogcFilters.enabled) {
+      igoFilters = ogcFilters.filters;
+    }
+    const ogcFilterWriter = new OgcFilterWriter();
+    const filterOrBox = ogcFilterWriter.buildFilter(igoFilters, extent, proj, ogcFilters.geometryName);
+    let filterOrPush = ogcFilterWriter.handleOgcFiltersAppliedValue(this.options, ogcFilters.geometryName);
+
+    let prefix = 'filter';
+    if (!filterOrPush) {
+      prefix = 'bbox';
+      filterOrPush = extent.join(',') + ',' + proj.getCode();
+    }
+
+    paramsWFS.xmlFilter = ogcFilters.advancedOgcFilters ? filterOrBox : `${prefix}=${filterOrPush}`;
+    let baseUrl = queryStringValues.find(f => f.name === 'getfeature').value;
+    const patternFilter = /(filter|bbox)=.*/gi;
+    baseUrl = patternFilter.test(paramsWFS.xmlFilter) ? `${baseUrl}&${paramsWFS.xmlFilter}` : baseUrl;
+    this.options.download = Object.assign({}, this.options.download, { dynamicUrl: baseUrl });
```

Comment:
Same here. Also, it appears that options.download.dynamicUrl is set twice.

Reply:
done